presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -164,10 +180,119 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         }
     }
 
+    public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint constraint)
+    {
+        requireNonNull(session, "session is null");
+        requireNonNull(tableHandle, "tableHandle is null");
+        requireNonNull(constraint, "constraint is null");
+        checkArgument(tableHandle instanceof IcebergTableHandle, "tableHandle not class IcebergTableHandle, but instead %s", tableHandle.getClass().getName());
+        IcebergTableHandle icebergTableHandle = (IcebergTableHandle) tableHandle;
+
+        if (constraint.getSummary().isNone()) {
+            return empty();
+        }
+
+        List<PartitionField> partitionFields = icebergTable.spec().fields();
+        List<Type> icebergPartitionTypes = partitionTypes(partitionFields);
+        List<IcebergColumnHandle> columnHandles = getColumns(icebergTable.schema(), typeManager);
+        Map<Integer, IcebergColumnHandle> idToColumnHandle = columnHandles.stream()
+                .collect(toUnmodifiableMap(IcebergColumnHandle::getId, identity()));
+
+        ImmutableMap.Builder<Integer, ColumnFieldDetails> idToDetailsBuilder = ImmutableMap.builder();
+        for (int index = 0; index < partitionFields.size(); index++) {
+            PartitionField field = partitionFields.get(index);
+            Type type = icebergPartitionTypes.get(index);
+            idToDetailsBuilder.put(field.sourceId(), new ColumnFieldDetails(field, idToColumnHandle.get(field.sourceId()), type, toPrestoType(type, typeManager), type.typeId().javaClass()));
+        }
+        Map<Integer, ColumnFieldDetails> idToDetails = idToDetailsBuilder.build();
+
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            TableScan tableScan = getTableScan(session, TupleDomain.all(), icebergTableHandle.getSnapshotId(), icebergTable).includeColumnStats();
+            Map<StructLikeWrapper, Partition> partitions = getPartitions(tableScan);
+
+            Partition summingPartition = null;
```

Comment:
Perhaps it's common, but I haven't seen "summing" used this way. It might be clearer call this just `summary` and name the methods like `updateSummaryMin`.

Reply:
Changed